### ISSUE TITLE
Add Examples to Discord Oauth provider.

### ISSUE THREAD

**Issue by AraHaan** [aspnet-contrib/AspNet.Security.OAuth.Providers#434]
I would like examples be added to some providers that do not currently add any yet as well other than just discord too.

Lets say for example I got an auth application setup to have discord redirect (from their page) to an api controller that basically redirects as ``https://mydomain.cf/authorize?code=[code that discord is adding the the url]`` With the scopes ``guilds guilds.join identify``

All I can find is adding the scopes and the client id and secret however I am not sure if I should add all 3 of them myself or just 2 of them because 1 is explicitly added for me?

Same for that redirect url how can the library just magically figure out what discord redirects the stuff too?

Or am I just doing things wrong?

**Comment by Neuroquila-n8fall** [comment]
An actual explanation would be very nice. I'm already sitting here for two hours figuring out how to use it properly.

**Comment by martincostello** [comment]
There's only so much documentation we can support for 60+ OAuth 2.0 providers.

There's some minimal documentation for the provider I've added recently at the links below, but how the OAuth provider itself works (e.g. what scopes do I need beyond the minimum) is for the providers' own documentation to deal with. We're basically just offering you C# bindings on top of the underlying API to configure appropriately for your needs.

  * https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/dev/docs/README.md
  * https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/dev/docs/discord.md

**Comment by Neuroquila-n8fall** [comment]
I had trouble with the redirect running into an unknown path while working with blazor and I really couldn't help but asking for a hint. I almost went crazy.
Long story short: I thought too complicated. Today I've started from scratch and now it works with the minimal example and a bog standard AccountController class. I guess it has something to do how blazor reacts to requests and since i couldn't wrap my head around that I went with the classical controller approach. Bingo. I mean this is what it has been designed for, right?

So, yes, the documentation is sufficient - I was just being stupid.
